### ISSUE TITLE
`prefer-const` false positive

### ISSUE THREAD

**Issue by modernserf** [eslint/eslint#13218]
**Tell us about your environment**

eslint.org/demo

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 11,
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "prefer-const": 2
    },
    "env": {
        "browser": false,
        "node": true
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let foo;

function bar () {
	console.log(foo)
}

bar();
foo = 1;
```

[reproducible on eslint.org/demo](https://eslint.org/demo#eyJ0ZXh0IjoibGV0IGZvbztcblxuZnVuY3Rpb24gYmFyICgpIHtcblx0Y29uc29sZS5sb2coZm9vKVxufVxuXG5iYXIoKTtcbmZvbyA9IDE7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJwcmVmZXItY29uc3QiOjJ9LCJlbnYiOnsiYnJvd3NlciI6ZmFsc2UsIm5vZGUiOnRydWV9fX0=)

**What did you expect to happen?**
No error

**What actually happened? Please include the actual, raw output from ESLint.**
```
1:5 - 'foo' is never reassigned. Use 'const' instead.
```

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by modernserf** [comment]
(this is solved if you write `let foo = undefined`, but then `no-undef-init` complains)

**Comment by mdjermanovic** [comment]
Hi @modernserf, thanks for the issue! 

For cases like this, you can use [ignoreReadBeforeAssign](https://eslint.org/docs/rules/prefer-const#ignorereadbeforeassign) option.

```js
/* eslint prefer-const: ["error", { ignoreReadBeforeAssign: true }] */

let foo;

function bar () {
	console.log(foo)
}

bar();
foo = 1;
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1jb25zdDogW1wiZXJyb3JcIiwgeyBpZ25vcmVSZWFkQmVmb3JlQXNzaWduOiB0cnVlIH1dICovXG5cbmxldCBmb287XG5cbmZ1bmN0aW9uIGJhciAoKSB7XG5cdGNvbnNvbGUubG9nKGZvbylcbn1cblxuYmFyKCk7XG5mb28gPSAxOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnsiYnJvd3NlciI6ZmFsc2UsIm5vZGUiOnRydWV9fX0=)
